README.md
![header](https://capsule-render.vercel.app/api?type=waving&color=00ffff&text=CollinIzDa&fontColor=FFF&fontSize=90&animation=fadeIn&fontAlignY=38&desc=pip%20install%20izda&descAlignY=51&descAlign=67&height=300&section=header)

<div align="center">
  <img src="https://collinizda.com/img/besen.png"></img>
</div>

<h1 align="center">
  Hello, welcome to the "izda" Repository!
</h1>

<p align="center">
<img alt="Version" src="https://img.shields.io/badge/version-1.0.3-blue.svg?cacheSeconds=2592000" />
<a href="https://pepy.tech/project/collinizda" target="_blank">
<img alt="Downloads" src="https://static.pepy.tech/personalized-badge/collinizda?period=total&units=international_system&left_color=grey&right_color=blue&left_text=Downloads" /></a>
<img src="https://img.shields.io/github/languages/top/CollinIzDa/pip-install-collinizda?style=flat-square" </a>
<img src="https://img.shields.io/github/license/CollinIzDa/pip-install-collinizda?style=flat-square" </a>
</p>

<p align="center">
<a href="https://github.com/CollinIzDa/pip-install-izda#---installation--">Installation</a> |
<a href="https://github.com/CollinIzDa/pip-install-collinizda/blob/main/Changelog.md">Changelogs</a> |
<a href="https://collinizda.com">Website</a>
</p>

> The package "**izda**" is just for fun and I coded it because I was bored.
> <br>
> It's not the best packageand my example is shit so don't expect too much

## - 📥 Installation -
```ssh
pip3 install izda
```

## - Functions in dropdown -
<details>
<summary>All functions with examples (Drop Down)</summary>

## - ❗ Title Function -
```py
import izda # Import the libary

izda.System.title("This is the Title!") # Change the title
```

## - Clear Function -
```py
import izda # Import the libary

izda.System.clear() # Clears the console
```

## - Write and Input -
```py
import izda # Import the libary

izda.Write.Print("Typewriter function", interval=0.05) # Typewriter print function
izda.Write.Print("Typewriter input function", interval=0.05, newLine=False) # Typewriter input animation
# The "interval" in the function is the speed and the "newLine=..." is to print a new line after the animation
# You can also use cursor=False/True if you want to show the command prompt cursor or not
```

## - All other -
```py
import izda # Import the libary

izda.System.init() # Allows you to use colors in print
izda.System.command("echo Hello") # Execute a command into the console
izda.System.enter(5) # Press enter 5 times to exit
```
</details>

---

## - 👤 Authors -
👤 Github: [@**CollinIzDa**](https://github.com/collinizda)<br>
👤 Website: [**CollinIzDa.com**](https://collinizda.com/)

## - 📍 Example -
```py
import izda # Import IzDa

def lol(): # Creating our example function
    izda.System.clear() # Clear the command prompt
    izda.System.title("Example") # Changes the title
    izda.System.init() # Allow us to write colors in the command prompt
    izda.System.command("mode 120, 25") # Execute a command
    inputa = izda.Write.Input(f'Write "Hi": ', interval=0.05) # Ask the user to input "Hi"
    if inputa != "Hi": # If he does not input "Hi"
        izda.Write.Print('I said you have to input "Hi"!', interval=0.05) # Say it to him
    else: # If he input "Hi" and did all right
        izda.Write.Print('Like this! Now press enter 5 times to exit.') # Say he have to press enter 5 times to exit
        izda.System.enter(5) # Execute the enter 5 times function

if __name__ == "__main__":
    lol() # Execute our created function on program run
```

---

🌟 **Enjoyed "izda"?** Consider dropping a star :D

**"izda" was created by CollinIzDa.**

Discord: CollinIzDa#1594

Website: https://collinizda.com/

![footer](https://capsule-render.vercel.app/api?type=waving&color=00ffff&height=200&section=footer)


Changelog.md
![header](https://capsule-render.vercel.app/api?type=waving&color=00ffff&text=IzDa&fontColor=FFF&fontSize=90&animation=fadeIn&fontAlignY=38&desc=Changelog&descAlignY=51&descAlign=55&height=300&section=header)

### 2023-03-02, v1.0.5
* ` Renamed the library to "izda"`
* ` Sorted my code better again`

### 2023-02-19, v1.0.2
* ` Sorted my code better`

### 2023-02-18, v1.0.1
* ` Directly found a bug and fixed it 💀`

### 2023-02-18, v1.0
* ` Just created this`
* ` Added title function to change the title from the command promt`
* ` Also added the enter function where you have to press enter
5 times to exit`
* ` Added clear function to clear the terminal`
* ` Added a typewriter print and input function`
* ` Exit is also there`
* ` A command function to execute a terminal command`

---

| 🌟Star This Repository If You Liked IzDa!|
|-------------------------------------------------|


izda/Write.py
# ================================== #
#        CollinIzDa's Library        #
#       https://collinizda.com       #
# ================================== #

# Copyright (c) CollinIzDa#1594
# That package was proudly coded by CollinIzDa#1594
# Its not done yet. If there any bugs or problems,
# create an issue on my github https://github.com/CollinIzDa/pip-install-collinizda/issues/new

#~~~ Imports ~~~#
import sys
sys.dont_write_bytecode = True

from time import sleep as _sleep
from pystyle import Cursor
#~~~ Imports ~~~#

"""
0 variables:
    Nothing here...

2 functions:
    print()    |    prints any text with a nice typewriter animation
    input()    |    writes an input text with a nice typewriter animation
"""

def Print(text: str, interval=0.05, show_cursor=True, newLine=False): # A fancy typewriter input function
    Cursor.HideCursor() # Hides the cursor for a better animation
    for i in text:  # Loop over the message
        # Print the one charecter, flush is used to force python to print the char
        print(i, end="", flush=True)
        _sleep(interval)  # Sleep a little before the next one
    if newLine:  # Check if the newLine argument is set to True
        print()  # Print a final newline to make it act more like a normal print statement
    if show_cursor:
        Cursor.ShowCursor() # Shows the cursor again

def Input(text: str, interval=0.05, show_cursor=True, newLine=False): # A fancy typewriter input function
    Cursor.HideCursor() # Hides the cursor for a better animation
    for i in text:  # Loop over the message
        # Print the one charecter, flush is used to force python to print the char
        print(i, end="", flush=True)
        _sleep(interval)  # Sleep a little before the next one
    if newLine:  # Check if the newLine argument is set to True
        print()  # Print a final newline to make it act more like a normal print statement
    if show_cursor:
        Cursor.ShowCursor() # Shows the cursor again
    return input() # returns input that we can input things in this text

izda/System.py
# ================================== #
#        CollinIzDa's Library        #
#       https://collinizda.com       #
# ================================== #

# Copyright (c) CollinIzDa#1594
# That package was proudly coded by CollinIzDa#1594
# Its not done yet. If there any bugs or problems,
# create an issue on my github https://github.com/CollinIzDa/pip-install-collinizda/issues/new

#~~~ Imports ~~~#
import sys
sys.dont_write_bytecode = True

from sys import stdout as _stdout
from ctypes import windll as _windll
from os import name as _name, system as _system
#~~~ Imports ~~~#

"""
3 variables:
    Windows    |    checks if the user is on the windows os or not
    Linux      |    checks if the user is on the linux os or not
    _ExitCode  |    Exit code for the _exit function

5 functions:
    clear()    |    clears the terminal
    title()    |    changes the console title
    write()    |    prints out words a little more fancier
    init()     |    initialize the terminal to allow the use of colors
    exit()     |    exits the terminal
    command()  |    execute a command into the console

NOTE: Bugs are possible because the library is not finished yet.
"""

Windows = _name == "nt"
Linux = _name == "posix"

def clear():
    if Windows:
        # If the os is windows
        _system("cls")
    elif Linux:
        # If the os is linux
        _system("clear")
    else:
        # I don't know what you are using
        print("\n"*120)

def title(_str):
    if Windows: # Checks if the user is using windows
        _windll.kernel32.SetConsoleTitleW(f"{_str}") # Update the title of the command promt

    elif Linux:
        _stdout.write(f"\x1b]0;{_str}\x07") # Update the title of the command promt
    
    else: # Bro what os are you using???
        pass # I do nothing

def init(): # initialize the terminal to allow the use of colors
    _system("")

def enter(times: str): # Press enter 5 times to exit
    return [input(i) for i in range(times, 0, -1)]  # Wait for 5 enter presses

def command(command: str): # Execute a terminal command
    return _system(command) # _system is os.system("the command e.g cls")

izda/__init__.py
# ================================== #
#        CollinIzDa's Library        #
#       https://collinizda.com       #
# ================================== #
PYTHONDONTWRITEBYTECODE = None
# Copyright (c) CollinIzDa#1594
# That package was proudly coded by CollinIzDa#1594
# Its not done yet. If there any bugs or problems,
# create an issue on my github https://github.com/CollinIzDa/pip-install-collinizda/issues/new

#~~~ Imports ~~~#
import sys
sys.dont_write_bytecode = True # I don't want bytecode lol

# Our modules
from .Write import *
from .System import *
#~~~ Imports ~~~#

#~~~ Variables ~~~#
__title__ = 'collinizda'
__author__ = 'CollinIzDa'
__license__ = 'GNU General Public License v3'
__copyright__ = 'Copyright (c) CollinIzDa 2023'
__version__ = '1.0.5'
#~~~ Variables End ~~~#

example.py
import izda # Import IzDa

def lol(): # Creating our example function
    izda.System.clear() # Clear the command prompt
    izda.System.title("Example") # Changes the title
    izda.System.init() # Allow us to write colors in the command prompt
    izda.System.command("mode 120, 25") # Execute a command
    inputa = izda.Write.Input(f'Write "Hi": ', interval=0.05) # Ask the user to input "Hi"
    if inputa != "Hi": # If he does not input "Hi"
        izda.Write.Print('I said you have to input "Hi"!', interval=0.05) # Say it to him
    else: # If he input "Hi" and did all right
        izda.Write.Print('Like this! Now press enter 5 times to exit.') # Say he have to press enter 5 times to exit
        izda.System.enter(5) # Execute the enter 5 times function

if __name__ == "__main__":
    lol() # Execute our created function on program run
